app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -191,71 +231,37 @@ export const snapKeyringBuilder = (
                 'AccountsController:setSelectedAccount',
                 internalAccount.id,
               );
-              await controllerMessenger.call('ApprovalController:showSuccess', {
-                header: [snapAuthorshipHeader],
-                title: t('snapAccountCreated') as string,
-                icon: IconName.UserCircleAdd,
-                message: [
-                  {
-                    name: 'SnapAccountCard',
-                    key: 'snapAccountCard',
-                    properties: {
-                      address,
-                    },
-                  },
-                  {
-                    name: 'Text',
-                    key: 'description',
-                    children: [
-                      t('snapAccountCreatedDescription') as string,
-                      ' ',
-                      learnMoreLink,
-                    ],
-                  },
-                ],
-              });
-            } catch (error) {
-              const subjectMetadata = controllerMessenger.call(
-                'SubjectMetadataController:getSubjectMetadata',
-                origin,
+
+              await showSuccess(
+                snapId,
+                t('snapAccountCreated') as string,
+                IconName.UserCircleAdd,
```

Comment:
is the Icon really something we need to pass as a param or can we simply default render the 'IconName.UserCircleAdd` inside the `showSuccess/showError`?

Reply:
I thought about this, but since success/error pages kind of lack "contexts" (of what is being shown), I think that letting the caller decide which icon is used, was better overall (but indeed, the less explicit parameters we have, the better it is IMO :D).

I also thoughts of using an option bags for those, so that we could have some sort of "optional" parameters, WDYT?

Reply:
I actually went for an options bag, IMO this looks nicer and could allow to have "optional" title or icon with this pattern too. BTW, the `ApprovalController` already supports it.